### PR TITLE
implement palette_changes and screen_melt options

### LINKED ISSUE #657: [Feature] Add more display options from PrBoom+
A few more display options to make the switch over from PrBoom+ would be very nice for someone like me to have. In specific, the option to disable palette changes for pain/bonus/powerups and disable the wipe screen effect are what are lacking from Woof! that are preventing me from making the full switch over.

That said I understand if this is outside of the scope of this project, just thought it was worth bringing up!

### PR DESCRIPTION
**Pull request by rfomin**
Fixes #657

### CODE REVIEW

File: src/d_main.c [hunk 1]
```diff
@@ -244,7 +245,7 @@ void D_Display (void)
     }
 
   // save the current screen if about to wipe
-  if ((wipe = gamestate != wipegamestate))
+  if ((wipe = gamestate != wipegamestate) && (screen_melt || strictmode))
```
Comment (fabiangreffrath) [inline src/d_main.c:248]: Should we introduce a `STRICTMODE` macro for this, similar to the one used in the menu code?

File: src/m_menu.c [hunk 2]
```diff
@@ -3953,33 +3961,39 @@ setup_menu_t gen_settings2[] = { // General Settings screen2
...
   {"Strict Mode", S_YESNO|S_LEVWARN, m_null, M_X,
    M_Y + general_strictmode*M_SPC, {"strictmode"}},
 
+  {"Show demo progress bar", S_YESNO, m_null, M_X,
+   M_Y + general_demobar*M_SPC, {"demobar"}},
+
+  {"On death action", S_CHOICE, m_null, M_X,
+   M_Y + general_death_action*M_SPC, {"death_use_action"}, 0, NULL, death_use_action_strings},
+
+  {"Palette changes", S_YESNO, m_null, M_X,
```
Comment (fabiangreffrath) [inline src/m_menu.c:3975]: Isn't this rather the first step towards Accessibility (although "Quality of Life" sounds fine as well).

File: src/d_main.c (hunk 1 above)
Comment (rfomin) [reply src/d_main.c:248]: I'm thinking maybe I should remove the `STRICTMODE` macro? Or write more generic one somehow.

File: src/m_menu.c (hunk 2 above)
Comment (rfomin) [reply src/m_menu.c:3975]: There is only two A11y options so far (maybe three, with bobbing). I think we can add another menu group if we implement more.

Comment (fabiangreffrath) [reply src/m_menu.c:3975]: Yes, right!

File: src/d_main.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/d_main.c:248]: I think the macro is fine. :shrug:

Comment (rfomin) [reply src/d_main.c:248]: Ok but how to name the new one? `NOTSTRICTMODE`? I'm confused here 😕

Comment (fabiangreffrath) [reply src/d_main.c:248]: Yes, sure, why not? 😉

**Pull request merged**